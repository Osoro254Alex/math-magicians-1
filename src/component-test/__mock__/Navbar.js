import React from 'react';
import { BrowserRouter as Router, NavLink } from 'react-router-dom';

class Navbar extends React.PureComponent {
  render() {
    return (
      <nav className="navbar">
        <Router>
          <NavLink className="link" to="/">
            Math Magicians
          </NavLink>
          <ul className="nav-items">
            <li className="nav-item">
              <NavLink className="link" to="/">
                Home
              </NavLink>
            </li>
            <li className="nav-item">
              <NavLink className="link" to="/Calculator">
                Calculator
              </NavLink>
            </li>
            <li className="nav-item">
              <NavLink className="link" to="/Quote">
                Quote
              </NavLink>
            </li>
          </ul>
        </Router>
      </nav>
    );
  }
}
export default Navbar;
