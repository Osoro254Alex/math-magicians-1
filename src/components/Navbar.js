import React from 'react';
import { NavLink } from 'react-router-dom';

class Navbar extends React.PureComponent {
  render() {
    return (
      <nav className="navbar">
        <NavLink className="link" to="/">
          Math Magicians
        </NavLink>
        <ul className="nav-items">
          <li className="nav-item">
            <NavLink className="link" to="/">
              Home
            </NavLink>
          </li>
          <li className="nav-item">
            <NavLink className="link" to="/Calculator">
              Calculator
            </NavLink>
          </li>
          <li className="nav-item">
            <NavLink className="link" to="/Quote">
              Quote
            </NavLink>
          </li>
        </ul>
      </nav>
    );
  }
}
export default Navbar;
