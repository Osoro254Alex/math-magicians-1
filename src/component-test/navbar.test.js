import React from 'react';
import { render, cleanup } from '@testing-library/react';
import Navbar from '../component-test/__mock__/Navbar';

afterEach(cleanup);

describe('render navbar component without error', () => {
  test('navbar should render correctly', () => {
    render(<Navbar />);
  });
});
