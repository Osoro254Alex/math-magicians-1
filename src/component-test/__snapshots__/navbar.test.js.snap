// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render navbar component without error matches snapshot for new Header 1`] = `
<nav
  className="navbar"
>
  <a
    aria-current="page"
    className="link active"
    href="/"
    onClick={[Function]}
  >
    Math Magicians
  </a>
  <ul
    className="nav-items"
  >
    <li
      className="nav-item"
    >
      <a
        aria-current="page"
        className="link active"
        href="/"
        onClick={[Function]}
      >
        Home
      </a>
    </li>
    <li
      className="nav-item"
    >
      <a
        className="link"
        href="/Calculator"
        onClick={[Function]}
      >
        Calculator
      </a>
    </li>
    <li
      className="nav-item"
    >
      <a
        className="link"
        href="/Quote"
        onClick={[Function]}
      >
        Quote
      </a>
    </li>
  </ul>
</nav>
`;
